.gitignore

```diff
@@ -1,4 +1,14 @@
 dist/
 node_modules/
-.vscode/
-npm-debug.log
\ No newline at end of file
+npm-debug.log
+
+# Created by https://www.gitignore.io/api/visualstudiocode
+
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
```

Comment:
.editorconfig does seem to be orthogonal to settings.json in many ways. As shown by the vscode source:

https://github.com/Microsoft/vscode/blob/master/.vscode/settings.json
https://github.com/Microsoft/vscode/blob/master/.editorconfig